### PR TITLE
Update roslyn analyzers

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: eng/config/rulesets/Shipping.ruleset [hunk 1]
```diff
@@ -110,6 +110,7 @@
     <Rule Id="RS0026" Action="Error" />
     <Rule Id="RS0027" Action="Error" />
     <Rule Id="RS0033" Action="None" /> <!-- Importing constructor should be [Obsolete] -->
+    <Rule Id="RS0040" Action="None" /> <!-- Defaultable types should have defaultable fields -->
```
Comment (sharwell) [inline eng/config/rulesets/Shipping.ruleset]: 📝 Enabling this is high-churn. I'm going to submit it as a separate pull request.

File: src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs [hunk 2]
```diff
@@ -374,7 +382,9 @@ private async ValueTask<DifferenceViewerPreview> CreateRemovedDocumentPreviewVie
             // the diff view would long have been dismissed by the time user edits the code)
             // resulting in crashes. Instead we create a new buffer from the same content.
             // TODO: We could use ITextBufferCloneService instead here to clone the original buffer.
+#pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task (containing method uses JTF)
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs]: Do you want to just pragma the entire file?

Comment (sharwell) [reply src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs]: I'm OK with this approach for now. We need to find a better long-term solution though.
